src/types/array.ts

```diff
@@ -1,16 +1,21 @@
 import { Runtype, Static, create } from '../runtype';
 import { ValidationError } from '../errors';
 
-interface Arr<E extends Runtype> extends Runtype<Static<E>[]> {
+type ArrayStaticType<E extends Runtype, RO extends boolean | undefined> = RO extends true ? ReadonlyArray<Static<E>> : Static<E>[]
+
+interface Arr<E extends Runtype, RO extends boolean | undefined> extends Runtype<ArrayStaticType<E, RO>> {
   tag: 'array';
   element: E;
+  isReadonly: RO;
```

Comment:
I wonder if the flag should actually be persisted beyond determining the type... it doesn’t have any runtime effect so maybe not?

Reply:
It's needed if we want `show` (and other reflect use cases) to be able to distinguish between the versions.

Is that a reasonable use case?

Reply:
Ah, yeah... I suppose so.